atlp-rwanda/phantom-fe-codebandits
# #CP27 Setup-Continuous-Integration-Quality-Assurance-pipeline

### What this PR do:

- [x] Add the GitHub actions integration on all branch for push
- [x] Add the workflow to deploy to Heroku
- [x] Add the workflow to deploy in production on push to main
- [x] Add Heroku pipeline apps
- [x] Add relevant badges to the repository READ-ME

### Description of task to be completed
Configure a continuous integration pipeline using GitHub Actions and Heroku. The continuous integration service should automatically run tests and other preliminary build steps required to ship the next iteration of the product

### How should this be manually tested
Using the GitHub actions- from now on every push will need to pass a check before merge
You can use the GitHub CLI for fast checkout and testing

```
// to install with CLI use
winget install github.cli
// for linux use the sudo and then authenticate with
gh auth login // optional
git clone repository
gh pr list
gh pr checkout <pr number without hashtag(#)>
gh browse 
// head to actions tabs
```
### Background context 
To serve the project on Heroku, I used the express packages for that

### Relevant Trello cards
#CP-27
